runner/materialize.go

```diff
@@ -122,32 +122,29 @@ func (m MaterializeRunner) Run() (types.CompletionWatcher, error) {
 		m.Logger.Infow("Creating Index", "name", m.ID.Name, "variant", m.ID.Variant)
 		vectorStore, ok := m.Online.(provider.VectorStore)
 		if !ok {
-			return nil, fmt.Errorf("cannot create index on non-vector store: %v", m.Online)
-		}
-		if !ok {
-			return nil, fmt.Errorf("cannot create index on non-vector type: %v", m.VType)
+			return nil, fferr.NewInternalError(fmt.Errorf("cannot create index on non-vector store: %v", m.Online))
 		}
 		_, err := vectorStore.CreateIndex(m.ID.Name, m.ID.Variant, vectorType)
 		if err != nil {
-			return nil, fmt.Errorf("create index error: %w", err)
+			return nil, err
 		}
 	}
 	m.Logger.Infow("Creating Table", "name", m.ID.Name, "variant", m.ID.Variant)
 	_, err = m.Online.CreateTable(m.ID.Name, m.ID.Variant, m.VType)
 	_, exists := err.(*fferr.DatasetAlreadyExistsError)
 	if err != nil && !exists {
-		return nil, fmt.Errorf("create table error: %w", err)
+		return nil, err
 	}
 	if exists && !m.IsUpdate {
-		return nil, fmt.Errorf("table already exists despite being new job")
+		return nil, fferr.NewDatasetAlreadyExistsError(m.ID.Name, m.ID.Variant, fmt.Errorf("table already exists despite being new job"))
```

Comment:
Should change to something like: "table x already exists" and print the table name

Reply:
There's not a clean way of getting the table name given the `OnlineStoreTable` interface only has `Get` and `Set` methods defined, but I'll change the error message.

Reply:
ah thats annoying